splunk-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,555 @@
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the VMs"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the VMs"
+      }
+    },
+    "splunkAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for Splunk admin"
+      }
+    },
+    "virtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the virtual network that the consumer wants to use"
+      }
+    },
+    "virtualNetworkNewOrExisting": {
+      "type": "string",
+      "defaultValue": "new",
+      "allowedValues": [
+        "new",
+        "existing"
+      ],
+      "metadata": {
+        "description": "Identifies whether to use new or existing Virtual Network"
+      }
+    },
+    "virtualNetworkExistingRGName": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Name of resource group of existing Virtual Network (if applicable)"
+      }
+    },
+    "virtualNetworkAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "Virtual network address CIDR"
+      }
+    },
+    "subnet1Name": {
+      "type": "string",
+      "defaultValue": "shsubnet",
+      "metadata": {
+        "description": "Subnet for the Search Head"
+      }
+    },
+    "subnet2Name": {
+      "type": "string",
+      "defaultValue": "idxsubnet",
+      "metadata": {
+        "description": "Subnet for the Indexers"
+      }
+    },
+    "subnet1Prefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "Search Head subnet CIDR"
+      }
+    },
+    "subnet2Prefix": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "Indexer subnet CIDR"
+      }
+    },
+    "subnet1StartAddress": {
+      "type": "string",
+      "defaultValue": "10.0.0.20",
+      "metadata": {
+        "description": "Search Head subnet start address"
+      }
+    },
+    "subnet2StartAddress": {
+      "type": "string",
+      "defaultValue": "10.0.1.20",
+      "metadata": {
+        "description": "Indexer subnet start address"
+      }
+    },
+    "sshFrom": {
+      "type": "string",
+      "defaultValue": "0.0.0.0/0",
+      "metadata": {
+        "description": "CIDR block from which SSH access is allowed (default is ssh access from anywhere)"
+      }
+    },
+    "forwardedDataFrom": {
+      "type": "string",
+      "defaultValue": "0.0.0.0/0",
+      "metadata": {
+        "description": "CIDR block from which forwarded data is allowed (default is data can be received from anywhere)"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "Size of the virtual machine"
+      }
+    },
+    "publicIPName": {
```

Comment:
I could be mistaken, but isn't that param necessary since UI template's PublicIpAddressCombo passes down both public IP `name` and `domainNameLabel`.
I agree, removing this param would be good.